molgenis/client.py

```diff
@@ -317,20 +387,148 @@ def get_attribute_meta_data(self, entity: str, attribute: str) -> dict:
 
         return response.json()
 
-    def upload_zip(self, meta_data_zip: str) -> str:
-        """Uploads a given zip with data and metadata"""
+    def get_meta(self, entity_type_id: str, expand: bool = False, abstract: bool = False):
+        """Similar to get_entity_meta_data(), but uses the newer Metadata API instead
+        of the REST API V1.
+        If expand is true, the metadata of the ref entities will be returned also.
+        If abstract is true, the metadata of the parent entity will be returned also.
+        """
+        response = self._session.get(
+            self._api_url + "metadata/" + quote_plus(entity_type_id) + "?flattenAttributes="+str(abstract),
+            headers=self._get_token_header(),
+        )
+
+        try:
+            response.raise_for_status()
+        except requests.RequestException as ex:
+            self._raise_exception(ex)
+
+        meta = response.json()["data"]
+
+        if expand:
+            for item in meta["attributes"]["items"]:
+                if "refEntityType" in item["data"]:
+                    ref_entity = item["data"]["refEntityType"]["self"][item["data"]["refEntityType"]["self"].rindex("/"):].replace("/", "")
+                    item["data"]["refEntityType"] = self.get_meta(ref_entity, abstract=True)
```

Comment:
I've checked this and because the default value of the expand parameter in self.get_meta = False it goes OK. When this parameter is included with value True, you indeed get an infinite loop for self-referencing attributes.